src/reading.rs

```diff
@@ -109,6 +109,18 @@ impl From<HashMap<String, FieldValue>> for Record {
     }
 }
 
+impl From<Record> for HashMap<String, FieldValue> {
+    fn from(record: Record) -> HashMap<String, FieldValue> {
+        record.map
+    }
+}
+
+impl From<&Record> for HashMap<String, FieldValue> {
+    fn from(record: &Record) -> HashMap<String, FieldValue> {
+        record.map.clone()
```

Comment:
I don't like the idea of having a clone here, ideally all the clones should be in user code.

So instead of implementing `From<&Record> for HashMap<String, FieldValue>` what should be implemented
is `AsRef<HashMap<String, FieldValue>> for Record` so that `&record` returns `&HashMap<String, FieldValue>` for free (no cloning required). And `AsMut<...> for Record`  should also be implemented to allow mutations.




https://doc.rust-lang.org/std/convert/trait.AsRef.html
https://doc.rust-lang.org/std/convert/trait.AsMut.html